tests/rules/test_asterisk_required.py

```diff
@@ -0,0 +1,199 @@
+    )
+
+
+def test_function_has_asterisk():
+    source_code = """def my_function(*, a):
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 0
+
+
+def test_function_no_params():
+    source_code = """def my_function():
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 0
+
+
+def test_function_missing_asterisk():
+    source_code = """def my_function(a):
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 1
+    assert occurrences[0] == Occurrence(
+        filename="my_file.py",
+        line_number=1,
+        rule_id=AsteriskRequiredRule.RULE_ID,
+        rule_label=AsteriskRequiredRule.RULE_LABEL,
+        function_name="my_function",
+    )
+
+
+def test_function_asterisk_too_late():
+    source_code = """def my_function(a, *, b):
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 1
+    assert occurrences[0] == Occurrence(
+        filename="my_file.py",
+        line_number=1,
+        rule_id=AsteriskRequiredRule.RULE_ID,
+        rule_label=AsteriskRequiredRule.RULE_LABEL,
+        function_name="my_function",
+    )
+
+
+def test_function_arg_with_defaults():
+    source_code = """def my_function(a=42):
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 1
+    assert occurrences[0] == Occurrence(
+        filename="my_file.py",
+        line_number=1,
+        rule_id=AsteriskRequiredRule.RULE_ID,
+        rule_label=AsteriskRequiredRule.RULE_LABEL,
+        function_name="my_function",
+    )
+
+
+def test_async_function_has_asterisk():
+    source_code = """async def my_function(*, a):
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 0
+
+
+def test_async_function_missing_asterisk():
+    source_code = """async def my_function(a):
+        pass
+    """
+
+    rule = AsteriskRequiredRule(filename="my_file.py", source_code=source_code)
+    occurrences = rule.check(source_code=source_code)
+
+    assert len(occurrences) == 1
+    assert occurrences[0] == Occurrence(
+        filename="my_file.py",
+        line_number=1,
+        rule_id=AsteriskRequiredRule.RULE_ID,
+        rule_label=AsteriskRequiredRule.RULE_LABEL,
+        function_name="my_function",
+    )
```

Comment:
question: Du hast jetzt immer diesen Occurrence-Block da drin, der großteils gleich aussieht. Kriegst Du das vielleicht etwas eleganter hin, so dass nur das da steht, was sich auch unterscheidet und getestet werden muss?

Reply:
Könnte man sicher irgendwie abstrahieren, aber lohnt die zusätzliche Komplexität?

Reply:
Wenn ich dadurch den Test schneller verstehe, finde ich schon:
```
def test_async_function_missing_asterisk():
    # arrange
    source_code = """async def my_function(a):
        pass
    """

    # act
    occurrences = AsteriskRequiredRule().check(source_code=source_code)

    # assert
    assert len(occurrences) == 1
    assert has_found(AsteriskRequiredRule) == True
```

Und während ich das Snippet gebastelt habe, habe ich mich gewundert, warum der Code in die Rule und in den Check reingegeben werden muss. Das wirkt komisch, finde ich.